okdata/sdk/sdk.py

```diff
@@ -28,30 +30,47 @@ def headers(self):
             headers["Authorization"] = f"Bearer {self.auth.access_token}"
         return headers
 
-    def post(self, url, data, **kwargs):
+    def post(self, url, data, retries=0, **kwargs):
         log.info(f"SDK:Posting resource to url: {url}")
-        result = requests.post(
+        session = self.prepared_request_with_retries(retries)
+        result = session.post(
             url, data=json.dumps(data), headers=self.headers(), **kwargs
         )
         result.raise_for_status()
         return result
 
-    def put(self, url, data, **kwargs):
+    def put(self, url, data, retries=0, **kwargs):
         log.info(f"SDK:Putting resource to url: {url}")
-        result = requests.put(
+        session = self.prepared_request_with_retries(retries)
+        result = session.put(
             url, data=json.dumps(data), headers=self.headers(), **kwargs
         )
         result.raise_for_status()
         return result
 
-    def get(self, url, **kwargs):
+    def get(self, url, retries=0, **kwargs):
         log.info(f"SDK:Getting resource from url: {url}")
-        result = requests.get(url, headers=self.headers(), **kwargs)
+        session = self.prepared_request_with_retries(retries)
+        result = session.get(url, headers=self.headers(), **kwargs)
         result.raise_for_status()
         return result
 
-    def delete(self, url, **kwargs):
+    def delete(self, url, retries=0, **kwargs):
         log.info(f"SDK:Deleting resource from url: {url}")
-        result = requests.delete(url, headers=self.headers(), **kwargs)
+        session = self.prepared_request_with_retries(retries)
+        result = session.delete(url, headers=self.headers(), **kwargs)
         result.raise_for_status()
         return result
+
+    @staticmethod
+    def prepared_request_with_retries(retries):
+        #  https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks/#retry-on-failure
+        retry_strategy = Retry(
+            total=retries, status_forcelist=[429, 500, 502, 503, 504]
+        )
```

Comment:
Burde vi sette `backoff_factor` også? Fra lenka over:
> This value is by default 0, meaning no exponential backoff will be set and retries will immediately execute. Make sure to set this to 1 in to avoid hammering your servers!

Burde vi også inkludere POST i `method_whitelist`?
> The HTTP methods to retry on. By default this includes all HTTP methods except POST because POST can result in a new insert. Modify this parameter to include POST because most API's I deal with don't return an error code and perform an insert in the same call. And if they do, you should probably issue a bug report.